### ISSUE TITLE
Projects/Rejected work: same user cannot submit application twice

### ISSUE THREAD

**Issue by PhilH** [AutarkLabs/open-enterprise#1633]
**Describe the bug**
Once work is rejected (for instance, with rating = "Needs rework"), the applicant cannot submit an application again. Only other users can.
NB: It seems weird to force rejection of applications submitted by the unsuccessful applicant, so I assume that it's a bug rather than a design decision. The fact that the new submission triggers the "Action impossible" rather than providing contextual information is not helping :)

**Mainnet or testnet?**
Rinkeby

**Organization**
oetest3

**To Reproduce**
1. Fund an issue as user A
2. Submit application as user B
3. Review/Accept application as user A
4. Submit work as user B
5. Reject work as user A
6. Try to submit application for the same issue as user B
7. "Action impossible"

**Expected behavior**
Let unsuccessful applicant submit applications again, since it's still possible after to reject their application

**Desktop (please complete the following information):**
 - OSX 10.13
 - Brave/Chrome

**Comment by jayalaves** [comment]
@PhilH Yes, this is most likely a bug. I've just come across the same problem now while running some tests and can confirm. I've also noticed that the error is gone if you switch to a different Ethereum address on MetaMask, without having to connect to a different GitHub account.

**Comment by PhilH** [comment]
Yes, this bug affects only users whose previous application was rejected. Sorry for not making it clearer, I tried ;)
